File: python/034_extractEachKth.py

def extractEachKth(inputArray, k):
   return [x for i, x in enumerate(inputArray) if (i + 1) % k != 0]
   
print(extractEachKth([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 3))

File: arcade/go/arcade-intro/08_Diving Deeper/034_extractEachKth.go
func extractEachKth(inputArray []int, k int) []int {
	current := 1
	result := []int{}

	for i := range inputArray {
		if current%k != 0 {
			result = append(result, inputArray[i])
		}
		current++
	}

	return result
}
